clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -4142,9 +4142,20 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
     if (SYCLStdArg) {
       SYCLStdArg->render(Args, CmdArgs);
       CmdArgs.push_back("-fsycl-std-layout-kernel-params");
+
+      unsigned version = llvm::StringSwitch<unsigned>(SYCLStdArg->getValue())
+                             .Cases("2017", "1.2.1", "121", "sycl-1.2.1", 2017)
+                             .Case("2020", 2020)
+                             .Default(0);
+      if (version == 2017)
+        CmdArgs.push_back("-DCL_SYCL_LANGUAGE_VERSION=121");
+      else if (version == 2020)
+        CmdArgs.push_back("-DCL_SYCL_LANGUAGE_VERSION=2020");
+
     } else {
       // Ensure the default version in SYCL mode is 1.2.1 (aka 2017)
       CmdArgs.push_back("-sycl-std=2017");
+      CmdArgs.push_back("-DCL_SYCL_LANGUAGE_VERSION=121");
```

Comment:
Please, move this to the front-end.
https://github.com/intel/llvm/blob/sycl/clang/lib/Frontend/InitPreprocessor.cpp#L466

I would also appreciate if commit new SYCL version support directly to llorg. See https://reviews.llvm.org/D72857 for reference.

Reply:
I've moved it into InitPreprocessor for this PR. Starting on the upstreaming to llvm.org